### PR TITLE
Rewrite GetExternalCRDPaths test helper to use go tools package

### LINKED ISSUE #2912: Addons webhooks tests failure when GOROOT is overriden
**Bug description**

tests sometimes fail locally in the addons directory on the main branch.

```
e.g.
 Failure [0.004 seconds]
ClusterbootstrapWebhook
/home/naadir/go/src/github.com/vmware-tanzu/tanzu-framework/addons/webhooks/clusterbootstrap_webhook_test.go:34
  Verify the logic of clusterbootstrap webhook
  /home/naadir/go/src/github.com/vmware-tanzu/tanzu-framework/addons/webhooks/clusterbootstrap_webhook_test.go:36
    should add defaults to the missing fields when the ClusterBootstrap CR has the predefined annotation [It]
    /home/naadir/go/src/github.com/vmware-tanzu/tanzu-framework/addons/webhooks/clusterbootstrap_webhook_test.go:63

    Unexpected error:
        <*errors.StatusError | 0xc00070b900>: {
            ErrStatus: {
                TypeMeta: {Kind: "", APIVersion: ""},
                ListMeta: {
                    SelfLink: "",
                    ResourceVersion: "",
                    Continue: "",
                    RemainingItemCount: nil,
                },
                Status: "Failure",
                Message: "admission webhook \"clusterbootstrap.mutating.vmware.com\" denied the request: tanzukubernetesreleases.run.tanzu.vmware.com \"v1.23.5---vmware.1-tkg.1-zshippable\" not found",
                Reason: "NotFound",
                Details: {
                    Name: "v1.23.5---vmware.1-tkg.1-zshippable",
                    Group: "run.tanzu.vmware.com",
                    Kind: "tanzukubernetesreleases",
                    UID: "",
                    Causes: nil,
                    RetryAfterSeconds: 0,
                },
                Code: 404,
            },
        }
        admission webhook "clusterbootstrap.mutating.vmware.com" denied the request: tanzukubernetesreleases.run.tanzu.vmware.com "v1.23.5---vmware.1-tkg.1-zshippable" not found
    occurred

    /home/naadir/go/src/github.com/vmware-tanzu/tanzu-framework/addons/webhooks/clusterbootstrap_webhook_test.go:76
```

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [X] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by randomvariable**
* Replaces shelling out with use of x/tools/go/packages
* Ensure dependencies are present in the system
* Compatible with users who have overrided GOROOT

Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

### Which issue(s) this PR fixes

Fixes #2912 

### Describe testing done for PR

`make test` passes

### Release note

```release-note
NONE
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: addons/test/testutil/test_helpers.go [hunk 1]
```diff
@@ -266,19 +266,21 @@ func getResource(decoder *yamlutil.YAMLOrJSONDecoder, mapper meta.RESTMapper, dy
...
-				"pkg", "mod", strings.TrimSuffix(string(depPath), "\n"), crdDir))
+			pkgs, err := packages.Load(packageConfig, dep)
+			if err != nil {
+				return nil, err
+			}
+			pkg := pkgs[0]
+			if pkg.Errors != nil {
+				return nil, pkg.Errors[0]
+			}
+			crdPaths = append(crdPaths, filepath.Join(pkg.Module.Dir, crdDir))
```
Comment (maralavi) [inline addons/test/testutil/test_helpers.go:283]: Seems like https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/pinniped/tanzu-auth-controller-manager/controllers/suite_test.go#L104 has a similar function ```getExternalCRDPaths()```

Comment (randomvariable) [reply addons/test/testutil/test_helpers.go:283]: yes, it will. thanks for pointing out.

Comment (randomvariable) [reply addons/test/testutil/test_helpers.go:283]: Done

**Review by maralavi — APPROVED** [review]

File: addons/controllers/suite_test_crd_dependencies.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+//go:build tools
+// +build tools
```
Comment (vijaykatam) [inline addons/controllers/suite_test_crd_dependencies.go:2]: Would the pinniped module also need a similar change?

**Comment by randomvariable** [comment]
Ready for merging.

**Pull request merged**